src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php

```diff
@@ -87,4 +101,23 @@ protected function getCheckoutHelper()
     {
         return Mage::helper('shopgate_cloudapi/frontend_checkout');
     }
+
+    /**
+     * Retrieves the passed in redirect URL from parameters
+     *
+     * @return false|string
+     */
+    private function getRedirect()
+    {
+        $req    = $this->getRequest();
+        $module = $req->getParam(self::REQUEST_REDIRECT_MODULE);
+        if ($module) {
+            $controller = $req->getParam(self::REQUEST_REDIRECT_CONTROLLER, 'index');
+            $action     = $req->getParam(self::REQUEST_REDIRECT_ACTION, 'index');
+
+            return Mage::getUrl("$module/$controller/$action", array('_secure' => true));
+        }
+
+        return false;
```

Comment:
just separating concerns, trying not to mix more logic in one method than needed. I am moving more towards cleaner functions that do one thing than cleaner calls. Currently the getRedirect just handles what it needs to. The checkout url is a fallback, but it's not a fallback for getRedirect function, it's a fallback for when there are no redirect params. Not sure if it makes sense. Shorter said, if at any point we would need code like this somewhere else, it would be easier to refactor it into a helper and make it backwards compatible here.

Reply:
acceptable or you want me to refactor it more?

Reply:
I guess that's fine